packages/shared/src/hooks/streaks/useReadingStreak.ts

```diff
@@ -1,27 +1,49 @@
-import { useQuery } from '@tanstack/react-query';
+import { useQuery, useQueryClient } from '@tanstack/react-query';
+import { useRef } from 'react';
 import { generateQueryKey, RequestKey, StaleTime } from '../../lib/query';
 import { getReadingStreak, UserStreak } from '../../graphql/users';
 import { useAuthContext } from '../../contexts/AuthContext';
-import { useStreakExperiment } from './useStreakExperiment';
+import { useActions } from '../useActions';
+import { ActionType } from '../../graphql/actions';
+import useDebounce from '../useDebounce';
 
 interface UserReadingStreak {
   streak: UserStreak;
   isLoading: boolean;
-  isEnabled: boolean;
+  shouldShowPopup: boolean;
+  checkReadingStreak: () => Promise<void>;
 }
 
 export const useReadingStreak = (): UserReadingStreak => {
   const { user } = useAuthContext();
-  const { shouldShowStreak } = useStreakExperiment();
   const { data: streak, isLoading } = useQuery(
     generateQueryKey(RequestKey.UserStreak, user),
     getReadingStreak,
-    { enabled: shouldShowStreak, staleTime: StaleTime.Default },
+    { enabled: true, staleTime: StaleTime.Default },
   );
+  const { checkHasCompleted } = useActions();
+  const hasReadToday =
+    new Date(streak?.lastViewAt).getDate() === new Date().getDate();
+  const userStreakQueryKeyRef = useRef<unknown[]>();
+  const queryClient = useQueryClient();
+
+  const [clearQueries] = useDebounce(async () => {
+    if (!hasReadToday && userStreakQueryKeyRef.current?.length > 0) {
+      await queryClient.invalidateQueries(userStreakQueryKeyRef.current);
+    }
+  }, 100);
 
   return {
     streak,
     isLoading,
-    isEnabled: shouldShowStreak,
+    checkReadingStreak: async () => {
```

Comment:
Why was this hook changed so much? Seems like new functionality to me vs what we had :)

Reply:
Same answer as the above https://github.com/dailydotdev/apps/pull/3091#discussion_r1601369413

One more functionality was added to it that was in useIncrementReadingRank which was deleted